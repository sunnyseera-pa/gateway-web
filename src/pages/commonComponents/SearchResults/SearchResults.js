import React from 'react';
import { Pagination, Row } from 'react-bootstrap';
import Loading from '../Loading';
import NoResults from '../NoResults';
import RelatedObject from '../relatedObject/RelatedObject';
import { DEFAULT_PROPS_SEARCH_RESULTS, PROP_TYPES_SEARCH_RESULTS } from './SearchResults.propTypes';

const SearchResults = ({
	results,
	type,
	count,
	search,
	data,
	sort,
	pageNumber,
	onPagination,
	maxResults,
	updateOnFilterBadge,
	isLoading,
	totalPages,
	errorMessage,
}) => (
	<>
		<div style={{ marginTop: '8px' }}>
			{!isLoading && (
				<>
					{sort && (
						<Row>
							<div className='text-right save-dropdown'>{sort}</div>
						</Row>
					)}
					{count <= 0 && (!errorMessage ? <NoResults type={type} searchString={search} /> : errorMessage({ search, type }))}
					{!results &&
						count > 0 &&
						data.map(item => {
							return (
								<RelatedObject key={item.id} data={item} activeLink={true} onSearchPage={true} updateOnFilterBadge={updateOnFilterBadge} />
							);
						})}
					{results && count > 0 && results(data)}
					{count > maxResults && (
						<Pagination>
							{new Array(Math.ceil(totalPages)).fill().map((value, i) => (
								<Pagination.Item key={i} active={i === pageNumber} onClick={() => onPagination(type, i * maxResults)}>
									{i + 1}
								</Pagination.Item>
							))}
						</Pagination>
					)}
				</>
			)}
			{!!isLoading && <Loading data-testid='loader' />}
		</div>
	</>
);

SearchResults.propTypes = PROP_TYPES_SEARCH_RESULTS;

SearchResults.defaultProps = DEFAULT_PROPS_SEARCH_RESULTS;

export default SearchResults;
