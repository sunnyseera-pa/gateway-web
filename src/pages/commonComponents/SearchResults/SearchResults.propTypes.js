import PropTypes from 'prop-types';

export const PROP_TYPES_SEARCH_RESULTS_TYPE = PropTypes.oneOf(['dataset', 'tool', 'project', 'paper', 'person', 'course', 'collection']);

export const PROP_TYPES_SEARCH_RESULTS = {
	results: PropTypes.func,
	type: PROP_TYPES_SEARCH_RESULTS_TYPE.isRequired,
	count: PropTypes.number,
	search: PropTypes.string,
	data: PropTypes.array.isRequired,
	sort: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
	pageNumber: PropTypes.number,
	onPagination: PropTypes.func,
	maxResult: PropTypes.number,
	updateOnFilterBadge: PropTypes.func,
	isLoading: PropTypes.bool,
	totalPages: PropTypes.number,
};

export const DEFAULT_PROPS_SEARCH_RESULTS = {
	count: 0,
	search: '',
	pageNumber: 1,
	maxResult: 40,
	totalPages: 0,
};
