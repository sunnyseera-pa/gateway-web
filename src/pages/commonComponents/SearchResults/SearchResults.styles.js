import { css } from '@emotion/react';

export const advancedSearch = css`
    display: flex;
    justify-content: center;
    margin-top: 10px;
`;
