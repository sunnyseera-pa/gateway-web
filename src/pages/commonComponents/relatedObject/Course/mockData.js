export default {
    _id: '5f96f3450d236c6cafdaa473',
    domains: ['test'],
    award: ['BSc'],
    courseOptions: [
        {
            flexibleDates: false,
            _id: '5f96f3450d236c6cafdaa476',
            startDate: '2020-11-12T00:00:00.000Z',
            studyMode: 'Full-time',
            studyDurationNumber: 5,
            studyDurationMeasure: 'Day(s)',
            fees: [
                {
                    _id: '5f96f3450d236c6cafdaa477',
                    feeDescription: 'Test',
                    feeAmount: 4,
                    feePer: 'Month',
                },
            ],
        },
        {
            flexibleDates: false,
            _id: '5f96f3450d236c6cafdaa476',
            startDate: '2020-11-12T00:00:00.000Z',
            studyMode: 'Full-time',
            studyDurationNumber: 5,
            studyDurationMeasure: 'Day(s)',
            fees: [
                {
                    _id: '5f96f3450d236c6cafdaa477',
                    feeDescription: 'Test',
                    feeAmount: 4,
                    feePer: 'Month',
                },
            ],
        },
    ],
    id: 3491286126459112,
    type: 'course',
    activeflag: 'active',
    title: 'Test A course',
    provider: 'Test A',
    description: 'Test A Test A Test A edited Test A',
    counter: 208,
    relatedresources: 3,
    updated: 1621465200000,
};
