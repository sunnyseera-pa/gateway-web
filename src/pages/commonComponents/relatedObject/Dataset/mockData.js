export default {
    categories: {
        programmingLanguage: [],
    },
    tags: {
        features: ['COVID-19', 'TRACKER', 'SAIL', 'COVID', 'MOBILE APP', 'CORONAVIRUS', 'SYMPTOM', 'NCS', 'National Core Study'],
        topics: [],
    },
    datasetfields: {
        publisher: 'ALLIANCE > TestSAIL',
        geographicCoverage: ['Great Britain'],
        physicalSampleAvailability: ['NOT AVAILABLE'],
        abstract: 'This is a test dataset abstract',
        ageBand: '15-150',
        metadataquality: {
            quality_score: 89.48,
        },
        phenotypes: [],
    },
    programmingLanguage: [],
    pid: 'c1f4b16c-9dfa-48e5-94ee-f0aa58c270e4',
    id: 401490453939233,
    datasetid: '78ce070c-84db-4676-81c5-0c672f4e12e3',
    type: 'dataset',
    activeflag: 'active',
    name: 'COVID-19 Symptom Tracker Dataset',
    description: 'This is a test dataset description',
    license: 'https://covid.joinzoe.com/privacy-notice',
    datasetv2: {
        summary: {
            publisher: {
                name: 'sail',
                logo: '',
                memberOf: 'ALLIANCE',
            },
        },
    },
    counter: 955,
    isCohortDiscovery: false,
    persons: [
        {
            firstname: 'Rich',
            lastname: 'Example',
        },
    ],
    latestUpdate: '2021-05-04T14:14:49.389Z',
    relatedresources: 1,
};
