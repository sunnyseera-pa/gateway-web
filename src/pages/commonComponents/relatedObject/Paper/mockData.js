export default {
    _id: '5f156b42577505d15e88f239',
    categories: {
        programmingLanguage: [],
    },
    tags: {
        features: ['Arbitrage'],
        topics: ['Blood'],
    },
    authors: [947228017269611],
    id: 18656838486747996,
    type: 'paper',
    name: 'This is a test paper',
    description: 'test paper description',
    activeflag: 'active',
    programmingLanguage: [],
    counter: 68,
    relatedresources: 5,
    persons: [
        {
            _id: '5e6f984a0a7300dc8f6fb196',
            id: 947228017269611,
            firstname: 'Test',
            lastname: 'Person',
        },
    ],
    updated: 1634511600000,
};
