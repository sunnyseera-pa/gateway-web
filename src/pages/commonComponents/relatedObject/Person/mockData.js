export default {
	_id: '5e544facbd427b6e9cd9059e',
	id: 900000011,
	lastname: 'Nei',
	bio: 'Jefferson Comprehensive Epilepsy Center',
	firstname: 'Maromi',
	type: 'person',
	tags: ['Neuroendocrinology', 'Epilepsy', 'Neuroscience'],
	activeflag: 'active',
	authors: [''],
	counter: 73,
	relatedresources: 0,
	persons: [],
	updated: 1621551600000,
};
