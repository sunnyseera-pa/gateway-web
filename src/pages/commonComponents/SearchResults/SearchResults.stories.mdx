import { Meta, Story, Canvas } from '@storybook/addon-docs';

import SearchResults from './SearchResults';
import { data } from './mockData';

<Meta
	title='Project/SearchResults'
	component={SearchResults}
	args={{
		data,
		count: 656,
		pageNumber: 3,
		totalPages: 10,
		type: 'datasets',
		search: 'datasets',
		updateOnFilterBadge: () => {},
		onPagination: () => {},
		maxResult: 40,
	}}
/>

<Canvas>
	<Story name='Basic'>{props => <SearchResults {...props} />}</Story>
</Canvas>
