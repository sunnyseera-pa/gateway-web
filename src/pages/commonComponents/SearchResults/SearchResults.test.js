import React from 'react';
import SearchResults from '.';
import { render } from '@testing-library/react';
import { data } from './mockData';

let wrapper;

const resultsProp = jest.fn();
const mockRelatedObject = jest.fn();

jest.mock('../relatedObject/RelatedObject', () => props => {
	mockRelatedObject(props);
	return <div />;
});

const props = {
	data,
	count: 656,
	pageNumber: 3,
	totalPages: 10,
	type: 'datasets',
	search: 'search term',
	updateOnFilterBadge: jest.fn(),
	onPagination: jest.fn(),
	maxResult: 40,
	isLoading: false,
	sort: 'Sort goes here...',
};

describe('Given the SearchResults component', () => {
	describe('When it is rendered', () => {
		beforeAll(() => {
			wrapper = render(<SearchResults {...props} isLoading={true} />);
		});

		it('Then matches the previous snapshot', () => {
			expect(wrapper.container).toMatchSnapshot();
		});

		it('Then contains a loader', () => {
			expect(wrapper.getByText('Loading...')).toBeTruthy();
		});

		describe('And it is no longer loading', () => {
			beforeAll(() => {
				wrapper.rerender(<SearchResults {...props} />);
			});

			it('Then matches the previous snapshot', () => {
				expect(wrapper.container).toMatchSnapshot();
			});

			it('Then doesnt contain a loader', () => {
				expect(wrapper.queryByText('Loading...')).toBeNull();
			});

			it('Then contains a sort', () => {
				expect(wrapper.getByText('Sort goes here...')).toBeTruthy();
			});

			it('Then has the correct number of results', () => {
				expect(mockRelatedObject).toHaveBeenCalledTimes(40);
			});

			describe('And page 2 is clicked', () => {
				beforeAll(() => {
					fireEvent.click(wrapper.getByText('2'));
				});

				it('Then calls onPagination with the correct args', () => {
					expect(props.onPagination).toHaveBeenCalledWith('dataset', 40);
				});
			});

			describe('And there is a cutom results function', () => {
				beforeAll(() => {
					mockRelatedObject.mockReset();

					wrapper.rerender(<SearchResults {...props} results={resultsProp} />);
				});

				it('Then calls the function with the correct data', () => {
					expect(resultsProp).toHaveBeenCalledWith(data);
				});
			});

			describe('And the count is zero', () => {
				beforeAll(() => {
					wrapper.rerender(<SearchResults {...props} count={0} />);
				});

				it('Then matches the previous snapshot', () => {
					expect(wrapper.container).toMatchSnapshot();
				});

				it('Then does not show results', () => {
					expect(mockRelatedObject).toHaveBeenCalledTimes(0);
				});

				it('Then shows a no results message', () => {
					expect(wrapper.getByText('We couldn’t find any datasets matching the search term ‘search term’')).toBeTruthy();
				});
			});

			describe('And there is only 1 page of results', () => {
				beforeAll(() => {
					wrapper.rerender(<SearchResults {...props} count={39} />);
				});

				it('Then does not show pagination', () => {
					expect(wrapper.queryByText('1')).toBeNull();
				});
			});
		});
	});
});
