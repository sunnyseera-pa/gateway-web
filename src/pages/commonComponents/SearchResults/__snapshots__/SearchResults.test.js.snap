// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the SearchResults component When it is rendered And it is no longer loading And the count is zero Then matches the previous snapshot 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="text-right save-dropdown"
    >
      Sort goes here...
    </div>
  </div>
  <div>
    <div
      class="mt-4 row"
    >
      <div
        class="gray800-14 text-center col"
      >
        <span>
           
          We couldn’t find any 
          datasets
           matching the search term ‘
          search term
          ’
           
        </span>
      </div>
    </div>
    <div
      class="mt-3 row"
    >
      <div
        class="gray800-14 text-center col"
      >
        <span>
           
          Try searching for ‘
          <a
            class="purple-14"
            href="/search?search=COVID-19"
          >
            COVID-19
          </a>
          ’ if you want to see examples of datasets
           
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Given the SearchResults component When it is rendered And it is no longer loading Then matches the previous snapshot 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="text-right save-dropdown"
    >
      Sort goes here...
    </div>
  </div>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <ul
    class="pagination"
  >
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
      >
        1
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
      >
        2
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
      >
        3
      </a>
    </li>
    <li
      class="page-item active"
    >
      <span
        class="page-link"
      >
        4
        <span
          class="sr-only"
        >
          (current)
        </span>
      </span>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
      >
        5
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
      >
        6
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
      >
        7
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
      >
        8
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
      >
        9
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
      >
        10
      </a>
    </li>
  </ul>
</div>
`;

exports[`Given the SearchResults component When it is rendered Then matches the previous snapshot 1`] = `
<div>
  <div
    style="margin-top: 30px;"
  >
    <div
      class="mt-2 row"
    >
      <div
        class="col"
      >
        <div
          class="rectangle"
        >
          <div
            class="gray800-14"
            style="text-align: center;"
          >
            <img
              class=""
              src=""
            />
          </div>
          <div
            class="gray800-14"
            style="text-align: center;"
          >
            Loading...
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
